client/components/modal/modal.service.js

```diff
@@ -207,6 +207,32 @@ app.factory('Modal', function($rootScope, $uibModal) {
         },
         model: {}
       }, 'modal-success', 'components/modal/form-modal.html');
+    },
+
+    confirmDelete: function(title, templateUrl, model, cb) {
+      var confirmDelete = openModal({
+        modal: {
+          dismissable: true,
+          title: title,
+          templateUrl: templateUrl,
+          submitFn: function(form, model) {
+            cb(model)
+            confirmDelete.close();
+          },
```

Comment:
So then it would be something like:

```javascript
cb(model).$promise.then(function(e) {
  confirmDelete.close(e);
}, function(err) {
  console.log(err);
});
```